psa-crypto/src/types/key.rs

```diff
@@ -897,12 +903,106 @@ impl TryFrom<Attributes> for psa_crypto_sys::psa_key_attributes_t {
+                    // Use values from https://www.secg.org/sec2-v2.pdf#subsection.A.2
+                    EccFamily::SecpK1 => match attributes.bits {
+                        192 | 224 | 256 => Ok(attributes.bits),
+                        _ => {
+                            error!("Requested key size is not supported ({})", attributes.bits);
+                            Err(Error::InvalidArgument)
+                        }
+                    },
+                    // SEC Koblitz curves over binary fields
+                    EccFamily::SectK1 => match attributes.bits {
+                        233 | 239 | 283 | 409 | 571 => Ok(attributes.bits),
+                        _ => {
+                            error!("Requested key size is not supported ({})", attributes.bits);
+                            Err(Error::InvalidArgument)
+                        }
+                    },
+                    // SEC random curves over binary fields
+                    EccFamily::SectR1 => match attributes.bits {
+                        233 | 283 | 409 | 571 => Ok(attributes.bits),
+                        _ => {
+                            error!("Requested key size is not supported ({})", attributes.bits);
+                            Err(Error::InvalidArgument)
+                        }
+                    },
+                    // Brainpool P random curves
+                    EccFamily::BrainpoolPR1 => match attributes.bits {
+                        192 | 224 | 256 | 320 | 384 | 512 => Ok(attributes.bits),
+                        _ => {
+                            error!("Requested key size is not supported ({})", attributes.bits);
+                            Err(Error::InvalidArgument)
+                        }
+                    },
+                    // Curve used primarily in France and elsewhere in Europe.
+                    EccFamily::Frp => match attributes.bits {
+                        256 => Ok(attributes.bits),
+                        _ => {
+                            error!("Requested key size is not supported ({})", attributes.bits);
+                            Err(Error::InvalidArgument)
+                        }
+                    },
+                    // Montgomery curves
+                    EccFamily::Montgomery => match attributes.bits {
+                        255 | 448 => Ok(attributes.bits),
+                        _ => {
+                            error!("Requested key size is not supported ({})", attributes.bits);
+                            Err(Error::InvalidArgument)
+                        }
+                    },
+                    _ => {
+                        // We don't (yet?) implement checks for other curve families
+                        Ok(attributes.bits)
+                    }
+                }
+            }
+            _ => {
+                // TO-DO We don't (yet?) implement checks for other types
+                Ok(attributes.bits)
```

Comment:
Doesn't this break any tests that involve symmetric primitives, like AEAD?

Reply:
Why would it? Functionality is the same as before for all types we haven't implemented checks in this function.

Reply:
Ah, yes, of course, apologies, for some reason I only read the "TODO" line and just assumed it returns an error 🤦🏻